### PR TITLE
ci: `test-readme` enhancements

### PR DESCRIPTION
**Pull request by pskfyi**
See #47, #59.

### CODE REVIEW

File: md/script/evalCodeBlocks.ts [hunk 1]
```diff
@@ -25,27 +25,33 @@ export async function evalCodeBlocks(
...
+    const firstChars = inlineCode.slice(
+      0,
+      consoleWidth - langLength - iconLength - 5,
+    );
 
     const message = firstChars.length < inlineCode.length
-      ? `${details.lang} ${firstChars}...`
-      : `${details.lang} ${firstChars}`;
+      ? `${colorIcon} ${details.lang} ${firstChars}...`
+      : `${colorIcon} ${details.lang} ${firstChars}`;
```
Comment (pskfyi) [inline md/script/evalCodeBlocks.ts:52]: New result output:

Screen Shot 2023-05-06 at 2 23 21 PM

Comment (pskfyi) [reply md/script/evalCodeBlocks.ts:52]: Should I still preserve the `\n` newline characters?

File: md/script/evalCodeBlocks.ts [hunk 2]
```diff
@@ -61,5 +69,9 @@ if (import.meta.main) {
     ? [[search, _replace]] as [string, string][]
     : undefined;
 
-  await evalCodeBlocks(filePath, replace);
+  const results = await evalCodeBlocks(filePath, replace);
+
+  results.forEach((result) => {
+    if (!(result instanceof Error) && !result.success) Deno.exit(1);
```
Comment (pskfyi) [inline md/script/evalCodeBlocks.ts:75]: Tested it in CI and it works 👍 

Screen Shot 2023-05-06 at 3 01 01 PM

Screen Shot 2023-05-06 at 3 01 39 PM

File: md/script/evalCodeBlocks.ts (hunk 1 above)
Comment (AustinArey) [reply md/script/evalCodeBlocks.ts:52]: > Should I still preserve the `\n` newline characters?

Let's not show the `\n`'s - I don't see a reason to indicate formatting here when the concern is the function of the code. Should we fold it in to this PR or create a TODO?

Comment (pskfyi) [reply md/script/evalCodeBlocks.ts:52]: I'll fold it in 👍  Thanks for the input!

Comment (pskfyi) [reply md/script/evalCodeBlocks.ts:52]: I went even further with my edit. Now all whitespace is collapsed to a single space, eliminating newlines but also erasing indentation. Here's the updated output at the same width as the previous screenshot.

Screen Shot 2023-05-07 at 7 21 46 PM

**Comment by pskfyi** [comment]
After rebasing on top of `main` CI is accurately surfacing a unit test added by this PR which was passing within the PR, which was branched off of an old `main`, but which fails in the context of the updated `main`. This is the exact reason that we want to [keep PRs in sync with the base branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch). I disabled this setting before because the GitHub UI unfortunately defaults to a merge commit rather than rebasing for this. I think we should try enabling for a while and seeing if it causes more issues than it catches.

Fix for the failing test is incoming.

File: md/script/evalCodeBlocks.ts (hunk 1 above)
Comment (AustinArey) [reply md/script/evalCodeBlocks.ts:52]: Excellent!

**Comment by AustinArey** [comment]
Updated branch via rebase. I see the force push above. Now `evalCodeBlocks()` test fails. 

I did `git fetch` then `git checkout test-readme-cross-platform` then `git pull` and `deno test -A md` ALL PASS

I am confused why it fails in the PR actions if I have the up to date code locally.

**Comment by pskfyi** [comment]
There was a compiler error introduced by the final commit because #49 [added the return type of `Promise<void>` to `evalCodeBlocks`](https://github.com/pskfyi/handy/blob/653bd4d7f9f21f4d445155fba068ac9ef8939232/md/script/evalCodeBlocks.ts#L12) (meaning it should return nothing or `undefined`) but the final commit here returned the [results](https://github.com/pskfyi/handy/blob/93d18ab159ed3dd15dc90255920bb8490d04a821/md/script/evalCodeBlocks.ts#L72) of `evaluateAll` from [earlier in the function](https://github.com/pskfyi/handy/blob/93d18ab159ed3dd15dc90255920bb8490d04a821/md/script/evalCodeBlocks.ts#L32). Updating the return type resolve the issue 👍

**Review by AustinArey — APPROVED** [review]
Nice work thanks for the merge conflict fix.

**Comment by pskfyi** [comment]
Awesome 🎉 

If you still have the failing code locally, you can run `deno task test-readme` and see that the `evalCodeBlocks` one fails. That's why it failed in CI in all 3 OS's.

Surprisingly, `deno lint` and `deno fmt --check` which we're running in CI don't surface the issue. I'm wondering if we should add a step to run [`deno check`](https://deno.com/manual@v1.31.2/getting_started/command_line_interface#type-checking-flags) in Ubuntu, which does surface these sorts of issues. Right now if we had such an error in a file not imported in a readme example, or only imported in a readme example which uses `no-eval`, then the compiler error would not be surfaced in CI.

**Pull request merged**